Mathlib/Analysis/Calculus/FirstDerivativeTest.lean

```diff
@@ -0,0 +1,237 @@
+/-
+Copyright (c) 2024 Bjørn Kjos-Hanssen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bjørn Kjos-Hanssen
+-/
+import Mathlib.Analysis.Calculus.MeanValue
+
+/-!
+# The First-Derivative Test
+
+We prove the first-derivative test in the strong form given on Wikipedia.
+
+The test is proved over the real numbers ℝ
+using `monotoneOn_of_deriv_nonneg` from [Mathlib.Analysis.Calculus.MeanValue].
+
+## Main results
+
+* `first_derivative_test_max`: Suppose `f` is a real-valued function of a real variable
+  defined on some interval containing the point `a`.
+  Further suppose that `f` is continuous at `a` and differentiable on some open interval
+  containing `a`, except possibly at `a` itself.
+
+  If there exists a positive number `r > 0` such that for every `x` in `(a − r, a)`
+  we have `f′(x) ≥ 0`, and for every `x` in `(a, a + r)` we have `f′(x) ≤ 0`,
+  then `f` has a local maximum at `a`.
+
+* `first_derivative_test_min`: The dual of `first_derivative_max`, for minima.
+
+## Tags
+
+derivative test, calculus
+-/
+
+
+
+/-!
+### Some facts about differentiability and continuity
+
+We prove a couple of auxiliary lemmas elaborating on facts such as
+"differentiable implies continuous",
+"an open interval is an open set", and "`fun x => -x` is antitone". -/
+
+
+/-- If `f` is differentiable on `(a,b)`, and `x ∈ (a,b)`, then `f` is differentiable at `x`.-/
+theorem differentiableOn_differentiableAt_Ioo.{u_1, u_2, u_3} {𝕜 : Type u_1}
+  [NontriviallyNormedField 𝕜] {E : Type u_2} [NormedAddCommGroup E] [NormedSpace 𝕜 E]
+    [LinearOrder E] [OrderClosedTopology E]
+    {a x b : E} (hab : x ∈ Set.Ioo a b)
+    {F : Type u_3} [NormedAddCommGroup F] [NormedSpace 𝕜 F]
+    {f : E → F}
+    (hd₀ : DifferentiableOn 𝕜 f (Set.Ioo a b)) :
+    DifferentiableAt 𝕜 f x := by
+  apply DifferentiableOn.differentiableAt
+  exact hd₀
+  refine IsOpen.mem_nhds ?hs.hs hab
+  apply isOpen_Ioo
+
+/-- If `f` is continuous at `b` and differentiable on `(a,b)` then `f` is
+  continuous on the half-open interval `(a,b]`. -/
+theorem continuous_Ioc.{u_1, u_2, u_3} {𝕜 : Type u_1} [NontriviallyNormedField 𝕜]
```

Comment:
The name is too generic and doesn't mention derivatives at all.

Reply:
Indeed, now I'm thinking there is no need for this to be a lemma. Converted to just a "have" statement.